src/battle_ai_main.c

```diff
@@ -529,9 +529,11 @@ static s16 AI_CheckBadMove(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
         // handle negative checks on non-user target
         // check powder moves
         if (TestMoveFlags(move, FLAG_POWDER) && !IsAffectedByPowder(battlerDef, AI_DATA->defAbility, AI_DATA->defHoldEffect))
-        {
             RETURN_SCORE_MINUS(20);
-        }
+
+        // Magic coat check.
+        if (TestMoveFlags(move, FLAG_MAGIC_COAT_AFFECTED) && gProtectStructs[battlerDef].bounceMove)
```

Comment:
this check doesn't actually do anything because the `bounceMove` field wont be set until the defender uses Magic Coat. AI calculations are computed before the player chooses a move. You could maybe change the second part of the `if` statement to: `gBattleMoves[predictedMove].effect == EFFECT_MAGIC_COAT`

Reply:
Noted, correcting..